flexinput/src/main/java/com/lytefast/flexinput/fragment/CameraFragment.kt

```diff
@@ -171,41 +172,24 @@ open class CameraFragment : PermissionsFragment() {
         ?.setOnClickListener { requestPermissionClick() }
   }
 
-  /**
-   * Some cameras don't properly set the [android.hardware.Camera.CameraInfo.facing] value.
-   * So here, if we fail, just try getting the first front facing camera.
-   */
-  private fun CameraView.tryStartCamera() {
-    Log.d(TAG, "Try starting the camera")
-    try {
-      if (isCameraOpened) {
-        stop()
-      }
-      start()
-    } catch (e: Exception) {
-      onCameraError(e, "Camera could not be loaded, try front facing camera")
-
-      try {
-        facing = CameraView.FACING_FRONT
-        start()
-      } catch (ex: Exception) {
-        onCameraError(e, "Camera could not be loaded")
-        Toast.makeText(context, R.string.camera_unknown_error, Toast.LENGTH_SHORT).show()
-      }
-    }
-  }
-
   override fun onPause() {
     handler.removeCallbacksAndMessages(null)
-    cameraView.stop()
+    cameraView.onPause()
     super.onPause()
   }
 
+  override fun onStop() {
+    cameraView.onStop()
+    super.onStop()
+  }
+
   protected open fun requestPermissionClick() {
     requestPermissions(object : PermissionsFragment.PermissionsResultCallback {
       override fun granted() {
         cameraView.post {
-          // #onResume will take care of this for us
+          Log.d(TAG, "Permission Granted")
+          cameraView.onStart()
+          cameraView.onResume()
```

Comment:
Just verifying this is intended, start & resume, right?

Reply:
Yup the life cycle is a bit wonky with CameraKitView for this scenario